src/main/java/com/hierynomus/smbj/share/DiskEntry.java

```diff
@@ -106,6 +107,19 @@ public void rename(String newName, boolean replaceIfExist, long rootDirectory) t
         this.setFileInformation(renameInfo);
     }
 
+    public void createHardlink( final String linkname ) 
```

Comment:
Can you add JavaDocs and format correctly?

Reply:
Seems to be ready.
Could you please tell me why CI for oracle java 9 fails?